invenio_rdm_migrator/streams/cache.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Migrator is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Record streams cache."""
+
+
+class ParentCache:
```

Comment:
we had some inconsistencies and I wanted to enforce the cache to be in a usable state so I put checks on `add`